### PR TITLE
Ebpf extend sync

### LINKED ISSUE #10741: Complement for `sync` monitoring
As discussed [here](https://github.com/netdata/netdata/issues/10725#issuecomment-794000014) and well defined [here](https://github.com/netdata/netdata/pull/10726#issuecomment-794008441), to bring the `sync` monitoring is not solving completely our user  problems,  we need to bring more `syscalls` monitoring to `netdata` to correct measure performance bottlenecks.

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes #10741 

This PR adds monitoring for more syscalls related to synchronization.

##### Component Name
ebpf.plugin

##### Test Plan

1 - Compile the Branch and start Netdata after this
2 - Compile this [example](https://github.com/netdata/netdata/files/6174427/tester.txt) and run it. When the software finalize you will have all charts with data.

##### Additional Information

This PR was tested on:

| Linux Distribution | kernels |
|------------------------|-------------|
| Slackware current | 5.11.2 |
| Slackware current | 5.10.24 |
| Slackware current | 5.4.95 |
| Ubuntu 18.04 | 5.4.66 |
| Ubuntu 18.04 | 4.15 |
| CentOS | 3.10.0 |

### CODE REVIEW

**Review by vlvkobal — CHANGES_REQUESTED** [review]

File: collectors/ebpf.plugin/ebpf.d/sync.conf [hunk 1]
```diff
@@ -12,3 +12,12 @@
...
     update every = 2
+
+# List of monitored syscalls 
+[syscalls]
+    sync = yes
+    msync = yes
+    fsync = yes
+    fdatasync = yes
+    syncfs = yes
+    sync_file_range = yes
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf.d/sync.conf:23]: As far as you added configuration for every syscall, you shouldn't create/update dimensions for disabled syscalls.

File: collectors/ebpf.plugin/ebpf_sync.c [hunk 2]
```diff
@@ -256,35 +287,24 @@ static void ebpf_sync_cleanup(void *ptr)
...
 {
     if (local_syscalls[NETDATA_SYNC_FSYNC_IDX].enabled || local_syscalls[NETDATA_SYNC_FDATASYNC_IDX].enabled)
-        ebpf_create_chart(NETDATA_EBPF_MEMORY_GROUP, NETDATA_EBPF_FILE_SYNC_CHART,
-                          "Monitor calls for <code>fsync(2)</code> and <code>fdatasync(2)</code>.",
-                          EBPF_COMMON_DIMENSION_CALL, NETDATA_EBPF_SYNC_SUBMENU, NULL,
-                          NETDATA_EBPF_CHART_TYPE_LINE, 21300,
-                          ebpf_create_global_dimension, &sync_counter_publish_aggregated[NETDATA_SYNC_FSYNC_IDX],
-                          2);
+        ebpf_create_sync_chart(NETDATA_EBPF_FILE_SYNC_CHART,
+                               "Monitor calls for <code>fsync(2)</code> and <code>fdatasync(2)</code>.", 21300,
```
Comment (ilyam8) [inline collectors/ebpf.plugin/ebpf_sync.c:313]: `<code>`

It doesn't work in the title, does it?

File: web/gui/dashboard_info.js [hunk 3]
```diff
@@ -1035,7 +1035,19 @@ netdataDashboard.context = {
...
+    'mem.file_sync': {
+        info: 'System calls for <a href="https://man7.org/linux/man-pages/man2/fsync.2.html" target="_blank">fsync() and fdatasync()</a> transfer all modified page caches for the files on disk devices. These calls block until the device reports that the transfer has been completed.'
+    },
+
+    'mem.memory_map': {
+        info: 'System calls for <a href="https://man7.org/linux/man-pages/man2/msync.2.html" target="_blank">msync()</a> which flushes changes made to the in-core copy of a file that was mapped.'
+    },
+
+    'mem.file_segment': {
+        info: 'System calls for <a href="https://man7.org/linux/man-pages/man2/msync.2.html" target="_blank">sync_file_range()</a> permits fine control when synchronizing the open file referred to by the file descriptor fd with disk. This system call is extremely dangerous and should not be used in portable programs.'
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply web/gui/dashboard_info.js]: Thank you I fixed it with the last commits.

File: collectors/ebpf.plugin/ebpf.d/sync.conf (hunk 1 above)
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf.d/sync.conf:23]: You are right, I brought two helpers that are fixing this.

File: collectors/ebpf.plugin/ebpf_sync.c (hunk 2 above)
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_sync.c:313]: At least on firefox it is working fine.

File: collectors/ebpf.plugin/ebpf_sync.c [hunk 4]
```diff
@@ -314,6 +336,13 @@ static void ebpf_create_sync_charts()
  */
 static void ebpf_sync_parse_syscalls()
 {
+    struct section *co = appconfig_get_section(&sync_config, NETDATA_SYNC_CONFIG_NAME);
+    if (co) {
+        struct config_option *cv;
+        for (cv = co->values; cv ; cv = cv->next) {
+        }
+    }
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_sync.c:344]: What for did you add this part?

Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_sync.c:344]: I used this to debug an information, I removed with the last commit. Thank you!

**Review by vlvkobal — APPROVED** [review]

**Review by underhood — APPROVED** [review]

**Pull request merged**